lang/ir/src/module_item/module_impl.rs

```diff
@@ -47,3 +58,28 @@ impl TryFrom<syn::ItemImpl> for ModuleImpl {
         })
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::ModuleImpl;
+
+    #[test]
+    fn impl_items_filtering() {
+        let item_impl: syn::ItemImpl = syn::parse_quote! {
+            impl Contract {
+                #[odra(init)]
+                pub fn constructor() {}
+
+                pub(crate) fn crate_public_fn() {}
+
+                pub fn public_fn() {}
+
+                fn private_fn() {}
+            }
+        };
+        let module_impl = ModuleImpl::try_from(item_impl).unwrap();
+
+        assert_eq!(module_impl.methods().len(), 4);
+        assert_eq!(module_impl.public_methods().len(), 2);
```

Comment:
`pub(crate)` is treated as private here, right?

Reply:
Yup, only pure `pub`s are public.